json_encoder.go

```diff
@@ -168,6 +172,46 @@ func (enc *jsonEncoder) AddMarshaler(key string, obj LogMarshaler) error {
 	return err
 }
 
+func (enc *jsonEncoder) addArrayBegin() {
+	enc.bytes = append(enc.bytes, '[')
+}
+
+func (enc *jsonEncoder) addArrayEnd() {
+	enc.bytes = append(enc.bytes, ']')
+}
+
+func (enc *jsonEncoder) addArraySep(i int) {
+	if i == 0 {
```

Comment:
I considered both approaches but choose the early return as it separates the validation from the logic and would probably be the preferable form if logic was > 1 line (i.e. textEncoder's addTime).  I'll keep this as-is unless @akshayjshah also feels strongly that it should be changed.